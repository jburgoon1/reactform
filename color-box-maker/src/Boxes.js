import React, {useState} from 'react'


const Boxes = ({width, height, backgroundColor, handleDelete}) =>{

 
return (
    <div style={{backgroundColor:backgroundColor, width:`${width}px`, height:`${height}px`}}>
        <button onClick={handleDelete}>X</button>
    </div>

)

}

export default Boxes;