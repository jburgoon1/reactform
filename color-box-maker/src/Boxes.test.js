import React from 'react'
import {render} from "@testing-library/react"
import Boxes from './Boxes'

it('renders without crashing', function (){
    render(<Boxes/>)
})

it('matches snapshot', function(){
    const {asFragment} = render(<Boxes/>)
    expect(asFragment()).toMatchSnapshot()
    
    })
it('matches snapshot', function(){
    const {asFragment} = render(<Boxes width='100px' height='100px' backgroundColor='blue'/>)
    expect(asFragment()).toMatchSnapshot()
    
    })