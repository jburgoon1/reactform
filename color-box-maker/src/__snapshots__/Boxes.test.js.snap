// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    style="background-color: blue;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
